test/tlv/TlvObjectTest.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * @license
+ * Copyright 2022 Project CHIP Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { TlvObject, TlvField, TlvOptionalField } from "../../src/tlv/TlvObject.js";
+import { TlvString } from "../../src/tlv/TlvString.js";
+import { TlvUInt, TlvUInt8 } from "../../src/tlv/TlvUInt.js";
+import { ByteArray } from "../../src/util/ByteArray.js";
+
+describe("TlvObject", () => {
+    const schema = TlvObject({
+        /** Mandatory field jsdoc */
+        mandatory: TlvField(1, TlvUInt8),
+
+        /** Optional field jsdoc */
+        optional: TlvOptionalField(2, TlvString()),
+    });
+
+    describe("encode", () => {
+        it("encodes an object with all fields", () => {
```

Comment:
Refactor to use a decode/encode test vector?

Reply:
Done in PR #20